modules/nextflow/src/main/groovy/nextflow/executor/BashFunLib.groovy

```diff
@@ -59,7 +59,10 @@ class BashFunLib {
                   ((i+=1))
                 fi
             done
-            ((\${#pid[@]}>0)) && wait \${pid[@]}
+            for p in "\${pid[@]}"
+            do
+                wait \$p
+            done
```

Comment:
Interesting, how does this version help propagating the error compared to `wait pid1 pid2 etc` ?

Reply:
`wait pid1 pid2 etc` doesn't return non-zero if any of the pids fail. `wait pid1` will return non-zero if the pid fails (and hence bomb due to `set -e`).

From what I can tell, `wait pid1 pid2 etc` isn't a valid use of `wait`: https://man7.org/linux/man-pages/man2/wait.2.html

Reply:
Nevermind, it is a valid use:
```
The exit
       status returned by the wait utility shall be the exit status of
       the process requested by the last pid operand.
```

From: https://www.man7.org/linux/man-pages/man1/wait.1p.html

Reply:
I see, make sense. But what happens when pid2 fails *before* pid1 terminates correctly? is still the error reported? by which wait?

Reply:
In that case, it'll wait for `pid1`, and then check `pid2`, see it failed, and nuke. From what I read, you actually can wait on procs in different orders than they finish, and it still works.

```
Therefore, the status of background processes was usually lost unless it terminated while wait was waiting for it. This could be  a
       serious	problem  when  a  job  that  was expected to run for a long time actually terminated quickly with a syntax or initialization error
       because the exit status returned was usually zero if the requested process ID was not found. This volume of  IEEE Std 1003.1-2001  requires
       the implementation to keep the status of terminated jobs available until the status is requested, so that scripts like:

	      j1&
	      p1=$!
	      j2&
	      wait $p1
	      echo Job 1 exited with status $?
	      wait $!
	      echo Job 2 exited with status $?

       work without losing status on any of the jobs.
```

- https://www.unix.com/man-page/linux/1/wait/

Reply:
ok, this is neat.